### DISCUSSION TITLE
Elfin EW11

### DISCUSSION THREAD

**Discussion by lempparisi** [veista/nilan#131] [Q&A]
Hi,

I have connected Elfin EW11 RS485 - Wifi device to my Nilan EC9.

I had to install Nilan integration manually by copying the files. HACS download gave me an error about the HA version, I didn't know how to fix it.

I'm trying to use TCP settings but I don't know the right port or the device ID?

I can find the device in my local network with IP address.
I can see that Serial Port has traffic.

I connected the device like so: (Wire colors are from my installation) (This cabel goes from Nilan EC9 to CTS602)
Red: +10vdc, connect here the EW11 +
Blue: GND, connect EW11 -
Yellow: +1Vdc, connect EW11 A
White: +3Vdc, connect EW11 B

**Comment by veista** [comment]
1. Im not sure if your bridge device is supported.

2. The current version does not work yet. Install older version via HACS

**Comment by veista** [comment]
Hi,

I have no experience on your bridge device, but with a quick google it is supposed to work.

<img width="592" alt="image" src="https://github.com/veista/nilan/assets/91916713/d92771c4-8869-426e-b891-95d2cb2a9afa">

This mode is what you need. Ports is what you want default 502, also Unit ID is what you set at both ends, default 30.

**Comment by lempparisi** [comment]
Hi,

Same here, no experience on this bridge. Just hopefully trying to get it to work.
Or I can also buy this one: https://www.waveshare.com/rs485-to-wifi-eth.htm
Maybe it is easier to get to work.

I tried to set in EW11 set up those ports to be the same in both (EW11 and NilanHA)
Communication settings:
![image](https://github.com/veista/nilan/assets/17859259/d56d7204-da69-4aee-8d4b-fbb8dc899f13)

SerialPort settings: These I got from discussion forum that posted installation process with waveshare rs485 to eth
![image](https://github.com/veista/nilan/assets/17859259/209c72bd-dd10-4eff-b9e0-da470598547b)

**Comment by lempparisi** [comment]
Update on this!

It is now working with Elfin EW11 and Nilan EC9, VP 18cek.

Finally found the correct settings! And now it is time for some automations!

**Reply by gllmlbrt** [reply]
Hey @lempparisi , I am late to this party, but could you clarify which settings on the EW11 ended up working ? Or is it what you posted just above. It is not clear.

**Reply by gllmlbrt** [reply]
Nevermind I figured it all out. but I won't tell also. Secret. :-)

**Comment by michaldzierzak-sys** [comment]
Thanks for the help! It's all up and running. If you have the same hardware/set, let me know and I'll send over the config.
